### PR TITLE
doc(ADR): icons

### PR DESCRIPTION
**Pull request by frassinier**
**Please check if the PR fulfills these requirements**

- [ ] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/design-system/adr/icons.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+Icons need to be hosted by Figma because they rely on many shapes, like atoms, in a systemic way. That's our Icon System.
+Having such a dedicated NPM package should only be the result of an optimization of that Figma output.
+In the meantime, Figma high-fidelity mockups must use the same optimized icons to inherit the current color, the consequence of that optimization too.
+
+## Solution
+
+Figma will stay the single source of truth.
+Figma will host the raw version, including the whole stack of shapes for each icon. 
+Those shapes will be based on the predefined atoms and will follow our guidelines for some consistency reasons.
```
Comment (Benrajalu) [inline packages/design-system/adr/icons.md:23]: https://www.figma.com/file/L1QUr28Y79ydAv05nQVmaA/Icon-Library?node-id=901%3A264

File: packages/design-system/adr/icons.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+Figma will expose the optimized version of those icons, on their own, as components.
+These components will be used by our product designers to compose the hi-fi mockups.
```
Comment (Benrajalu) [inline packages/design-system/adr/icons.md:26]: https://www.figma.com/file/L1QUr28Y79ydAv05nQVmaA/Icon-Library?node-id=2064%3A17133 (automated)

File: packages/design-system/adr/icons.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+Each React component get a hashed name based on its content `[contenthash].js`.  
+If you have already fetched the file, the web browser cache can reuse it.  
+If not, it will fetch a couple of kilobytes for once. 
+
+Now, you can say "OK, but what if my version of the package `@talend/icons` changes?"  
+That's why we introduce the service worker, since, if the version changes, that implies a new URL to reach the CDN. 
+And a new URL is a new cache key for your web browser.
+
+The package `@talend/icons` won't expose a generic `<Icon />` React component. 
+This component will be part of the Design System package.
```
Comment (Benrajalu) [inline packages/design-system/adr/icons.md:80]: In effect:

```
@import { Icon } from "@talend/design-system";

...

<Icon size="S" name="carret-down" />
// Where
// Size = "XS" | "S" | "M" | "L";
// Name = IconName<T extends Size>;
// Basically TS will let you know if the icon you are calling is available in that specific size.
```

**Review by tlnd-mhuchet — APPROVED** [review]
👍

**Comment by frassinier** [comment]
Won't do, UX is not optimized with real use cases.

**Pull request closed without merging**